public/app/features/dashboard-scene/settings/VariablesEditView.tsx

```diff
@@ -106,16 +148,83 @@ export class VariablesEditView extends SceneObjectBase<VariablesEditViewState> i
   };
 
   public onEdit = (identifier: string) => {
-    return 'not implemented';
+    const variableIndex = this.getVariableIndex(identifier);
+    if (variableIndex === -1) {
+      console.error('Variable not found');
+      return;
+    }
+    this.setState({ editIndex: variableIndex, originalVariableState: { ...this.getVariables()[variableIndex].state } });
+  };
+
+  public onTypeChange = (type: EditableVariableType) => {
+    // Find the index of the variable to be deleted
+    const variableIndex = this.state.editIndex ?? -1;
+    const { variables } = this.getVariableSet().state;
+    const variable = variables[variableIndex];
+
+    if (!variable) {
+      // Handle the case where the variable is not found
+      console.error('Variable not found');
+      return;
+    }
+
+    const { name, label } = variable.state;
```

Comment:
I was testing a bit and found a bug when the user changes the type to textbox and then goes back 🤔 


https://github.com/grafana/grafana/assets/239999/6cd72dd1-1f21-467d-8d49-8bde61958c5a